src/sentry/api/endpoints/project_release_repositories.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+
+from sentry.api.bases.project import ProjectEndpoint, ProjectReleasePermission
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.api.serializers import serialize
+
+from sentry.models import Release, ReleaseCommit, Repository
+
+
+class ProjectReleaseRepositories(ProjectEndpoint):
+
+    permission_classes = (ProjectReleasePermission,)
+
+    def get(self, request, project, version):
+        """
+        Retrieve Project Repositories from a Release
+        ````````````````````````````
+
+        This endpoint is used in the commits and changed files tab of the release details page
+
+        :pparam string organization_slug: the slug of the organization the
+                                          release belongs to.
+        :pparam string project_slug: the slug of the project to retrieve the
+                                     release of.
+        :pparam string version: the version identifier of the release.
+        :auth: required
+        """
+
+        try:
+            release = Release.objects.get(
+                organization_id=project.organization_id, projects=project, version=version
+            )
+        except Release.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        release_commits = ReleaseCommit.objects.filter(release=release).select_related("commit")
```

Comment:
If you only need the repositories, you could use `.select_related("commit__repository")` to get everything in fewer queries.

Reply:
Thank you for pointing this out, but unfortunately, it doesn't seem to have an association.

Reply:
I'd try `commit__repository_id`

Reply:
nvm that doesn't work because it would not be an extra join